src/main/java/gov/cabinetofice/gapuserservice/service/encryption/AESService.java

```diff
@@ -0,0 +1,125 @@
+
+    @Autowired
+    private Sha512Service sha512Service;
+
+    public AESService() throws Exception {
+        KeyGenerator keyGenerator = KeyGenerator.getInstance(KEY_SELECTOR);
+        keyGenerator.init(AES_KEY_SIZE);
+
+        secretKey = keyGenerator.generateKey();
+
+    }
+
+    public Optional<EncryptedResponseDto> encrypt(String message) {
+        try {
+            byte[] initVector = new byte[GCM_IV_LENGTH];
+            GenerateInitVector(initVector);
+
+            Cipher cipher = Cipher.getInstance(CIPHER_SELECTOR);
+            SecretKeySpec keySpec = new SecretKeySpec(secretKey.getEncoded(), KEY_SELECTOR);
+            GCMParameterSpec gcmSpec = new GCMParameterSpec(GCM_TAG_LENGTH , initVector);
+            cipher.init(Cipher.ENCRYPT_MODE, keySpec, gcmSpec);
+            byte[] cipherText = cipher.doFinal(message.getBytes());
+
+            EncryptedResponseDto response = new EncryptedResponseDto();
+            response.setEncryptedText(DatatypeConverter.printHexBinary(cipherText));
+            response.setIV(initVector);
+            response.setPrivateKey(secretKey);
+            response = signingService.signMessage(response);
+            response.setShaSum(sha512Service.getSHA512SecurePassword(response.getEncryptedText() + response.getSignature()));
+            return Optional.of(response);
+        }
+        catch (NoSuchAlgorithmException
+                | InvalidKeyException
+                | InvalidAlgorithmParameterException
+                | NoSuchPaddingException
+                | BadPaddingException
+                | IllegalBlockSizeException e) {
+            e.printStackTrace();
+        }
+        return Optional.empty();
+    }
+
+    private void GenerateInitVector(byte[] initVector) {
+        SecureRandom secureRandom = new SecureRandom();
+        secureRandom.nextBytes(initVector);
+    }
+
+    public Optional<String> decrypt(EncryptedResponseDto response) {
+        var shaSum  = response.getEncryptedText() + response.getSignature();
+        shaSum = sha512Service.getSHA512SecurePassword(shaSum);
+
+        if(signingService.verify(response) && response.getShaSum().equals(shaSum)) {
+            try {
+                Cipher cipher = Cipher.getInstance(CIPHER_SELECTOR);
+                SecretKeySpec spec = new SecretKeySpec(secretKey.getEncoded(), KEY_SELECTOR);
+                GCMParameterSpec gcmSpec = new GCMParameterSpec(GCM_TAG_LENGTH * 8, response.getIV());
+                cipher.init(Cipher.DECRYPT_MODE, spec, gcmSpec);
+                return Optional.of(new String(cipher.doFinal(DatatypeConverter.parseHexBinary(response.getEncryptedText()))));
+            }
+            catch (NoSuchAlgorithmException
+                    | InvalidKeyException
+                    | InvalidAlgorithmParameterException
+                    | NoSuchPaddingException
+                    | BadPaddingException
+                    | IllegalBlockSizeException e) {
+                e.printStackTrace();
```

Comment:
Should catch this exception and log it using the Logger mechanism rather than just dumping the stack trace